Compiler/Contract/Helpers.cs

```diff
@@ -513,13 +513,22 @@ public static void CheckValueTypeClone(ResolveResult resolveResult, Expression e
                     writeClone = true;
                 }
 
+                var memberResult = resolveResult as MemberResolveResult;
+                bool isIMethod = memberResult?.Member is IMethod;
+                bool isInvocationResult = memberResult is InvocationResolveResult;
+                // check below fixes issue with
+                // delegate with mutable struct as a return value
+                // ex: delegate Vector2 ScratchHandler();
+                if (isIMethod  && !(isInvocationResult)){
```

Comment:
```suggestion
                if (isIMethod && !(isInvocationResult)){
```
Probably, we need to integrate code styles in bridge, in different ttask/pr of cource